### PR TITLE
Documented Cisco ACI Plugin Deployment Instructions

### PR DESCRIPTION
**Pull request by sunithasomanna**
Drafted instructions for deploying Cisco ACI plugin in the readme.

### CODE REVIEW

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
This can be updated only after the ODIMRA deployment docs are updated to support kubernetes. We cant have the base product on docker and plugin on kubernetes .

File: README.md [hunk 1]
```diff
@@ -1,50 +1,362 @@
...
+   $ sudo chown odimra:odimra /var/log/aciplugin_logs
+   ```
+
+5. On the deployment node, copy the Cisco ACI plugin configuration file to `~/plugins/aciplugin`.
+
+   ```
+   $ cp ~/ODIM/odim-controller/helmcharts/aciplugin/aciplugin-config.yaml ~/plugins/aciplugin
+   ```
+
+5. Open the Dell plugin configuration YAML file.
```
Comment (sukhwinder20) [inline README.md]: Change Dell to ACI

File: README.md [hunk 2]
```diff
@@ -1,50 +1,362 @@
...
+
+   ```
+   $ vi ~/plugins/aciplugin/aciplugin-config.yaml
+   ```
+
+   **Sample aciplugin-config.yaml file:**
```
Comment (sukhwinder20) [inline README.md:73]: Sample file is not correct missing other parameters

File: README.md [hunk 3]
```diff
@@ -1,50 +1,362 @@
...
+
+   2. Run the following command to create `aciplugin` Helm package at `~/plugins/aciplugin`:
+
+      ```
+      $ helm package aciplugin -d ~/plugins/aciplugin
+      ```
+
+      The Helm package for the Cisco ACI plugin is created in the tgz format.
+
+8. Save the Dell plugin Docker image on the deployment node at `~/plugins/aciplugin`.
```
Comment (sukhwinder20) [inline README.md]: Change the Dell to ACI

File: README.md [hunk 4]
```diff
@@ -1,50 +1,362 @@
...
+    "Links":{
+         "ConnectionMethod": {
+            "@odata.id": "/redfish/v1/AggregationService/ConnectionMethods/{ConnectionMethodId}"
+         }
+   }
+   }' \
+ 'https://{odim_host}:30080/redfish/v1/AggregationService/AggregationSources'
+   ```
+   
+   <blockquote>
```
Comment (sukhwinder20) [inline README.md:283]: blockquote needs to be removed

File: README.md [hunk 5]
```diff
@@ -1,50 +1,362 @@
...
+
+      -   Links to all the added plugin managers.
+
+   2. To identify the plugin id of the added plugin, perform HTTP `GET` on each manager link in the response. 
+
+      The JSON response body for a plugin manager has `Name` as the plugin name.
+      Example:
+      The JSON response body for the URP plugin manager has `Name` as `URP`.
+
+      **Sample response \(URP manager\)**
```
Comment (sukhwinder20) [inline README.md]: Change example with ACI

File: README.md [hunk 6]
```diff
@@ -1,50 +1,362 @@
...
+   ```
+   $ echo -n '{odim_username}:{odim_password}' | base64 -w0
+   ```
+   
+   Replace `{base64_encoded_string_of_[odim_username:odim_password]}` with the generated base64 encoded string in the curl command. You will receive:
+   
+   -   An HTTP `202 Accepted` status code.
+   -   A link to the task monitor associated with this operation in the response header.
+   
+   To know the status of this task, perform HTTP `GET` on the `taskmon` URI until the task is complete. If the plugin is added successfully, you will receive an HTTP `200 OK` status code.
```
Comment (sukhwinder20) [inline README.md]: Format needs to be changed here

File: README.md [hunk 7]
```diff
@@ -1,50 +1,362 @@
...
+2. Create a plugin configuration file called `<plugin-name>_nginx_server.conf`: 
+
+   ```
+   $ vi <plugin-name>_nginx_server.conf
+   ```
+
+   Example:
+
+   ```
+   $ vi grfplugin_nginx_server.conf
```
Comment (sukhwinder20) [inline README.md]: change grf to aci

File: README.md [hunk 8]
```diff
@@ -1,50 +1,362 @@
...
+
+   3. Check in the JSON response of the plugin manager, if: 
+
+      -    `State` is `Enabled` 
+
+      -    `Health` is `Ok` 
+
+      For more information, refer to "Managers" in [Resource Aggregator for Open Distributed Infrastructure Management™ API Reference and User Guide](https://github.com/ODIM-Project/ODIM/tree/development/docs).
+
+## Configuring proxy server for a plugin version
```
Comment (sukhwinder20) [inline README.md:361]: This section needs to be changed fully

**Review by sukhwinder20 — CHANGES_REQUESTED** [review]
Please address the changes

File: README.md (hunk 2 above)
Comment (sukhwinder20) [reply README.md:73]: Sample file is not correct ,Kindly refer aci pdf guide

Comment (sukhwinder20) [reply README.md:73]: Also give the description for each parameter

File: README.md [hunk 9]
```diff
@@ -1,50 +1,1323 @@
...
+   $ sudo docker save aciplugin:1.0 -o ~/plugins/aciplugin/aciplugin.tar
+   ```
+9. If it is a three-node cluster configuration, log in to each cluster node and [configure proxy server for the plugin](#configuring-proxy-server-for-a-plugin-version). 
+
+   Skip this step if it is a one-node cluster configuration.
+
+10. Navigate to the `/ODIM/odim-controller/scripts` directory on the deployment node.
+
+    ```
+    $ cd ~/ODIM/odim-controller/scripts
```
Comment (sukhwinder20) [inline README.md]: 8th step in the pdf file needs to be run here "Update the following parameters in the kube_deploy_nodes.yaml file to their corresponding values:"

File: README.md [hunk 10]
```diff
@@ -1,50 +1,1323 @@
...
+      Example:
+      The JSON response body for the Cisco ACI plugin manager has `Name` as `CiscoACI`.
+
+      **Sample response**
+
+      ```
+      {
+              "@odata.context":"/redfish/v1/$metadata#Manager.Manager",
+              "@odata.etag":"W/\"AA6D42B0\"",
+              "@odata.id":"/redfish/v1/Managers/536cee48-84b2-43dd-b6e2-2459ac0eeac6",
```
Comment (sukhwinder20) [inline README.md]: sample response is:

{
    "@odata.context": "/redfish/v1/$metadata#Manager.Manager",
    "@odata.id": "/redfish/v1/Managers/fb40f2dc-0c6d-4464-bc98-fea775adbbb9",
    "@odata.type": "#Manager.v1_10_0.Manager",
    "FirmwareVersion": "v1.0.0",
    "Id": "fb40f2dc-0c6d-4464-bc98-fea775adbbb9",
    "Links": {
        "ManagerForSwitches": [
            {
                "@odata.id": "/redfish/v1/Fabrics/fb40f2dc-0c6d-4464-bc98-fea775adbbb9:1/Switches/af10c386-68d5-45aa-b3c3-431e3e4c3647:101"
            },
            {
                "@odata.id": "/redfish/v1/Fabrics/fb40f2dc-0c6d-4464-bc98-fea775adbbb9:1/Switches/668f20cf-b6e7-4ded-a180-bf8e33dc18fc:102"
            },
            {
                "@odata.id": "/redfish/v1/Fabrics/fb40f2dc-0c6d-4464-bc98-fea775adbbb9:1/Switches/7d5a25b3-3ac4-49f4-a929-243b5b97bba0:201"
            }
        ],
        "ManagerForSwitches@odata.count": 3
    },
    "ManagerType": "Service",
    "Name": "ACI",
    "Status": {
        "Health": "OK",
        "State": "Enabled"
    },
    "UUID": "fb40f2dc-0c6d-4464-bc98-fea775adbbb9"
}

File: README.md [hunk 11]
```diff
@@ -1,50 +1,1323 @@
...
+
+
+When deleting fabric entities, ensure to delete them in the following order:
+
+1.  Zone of endpoints
+
+2.  Endpoints
+
+3.  Zone of zones
+
```
Comment (sukhwinder20) [inline README.md:462]: 4th point should be default zone

File: README.md [hunk 12]
```diff
@@ -1,50 +1,1323 @@
...
+   -H "X-Auth-Token:{X-Auth-Token}" \
+    -d \
+'{
+  "Name": "HPE-AddressPool-ZoneofZone",
+"Description": "VLANRange for creating domain",
+"Ethernet": {
+"IPv4": {
+"VLANIdentifierAddressRange": {
+"Lower": 100,
+"Upper": 200
```
Comment (sukhwinder20) [inline README.md:513]: Arrange the brackets properly

File: README.md [hunk 13]
```diff
@@ -1,50 +1,1323 @@
...
+"Description":"Default Zone",
+"ZoneType":"Default"
+}
+```
+
+**Request parameters**
+
+| Parameter   | Type                  | Description                                                  |
+| ----------- | --------------------- | ------------------------------------------------------------ |
+| Name        | String (optional)     | Name for the zone.<br />**NOTE**: Ensure that there are no spaces. |
```
Comment (sukhwinder20) [inline README.md:729]: <br /> needs to be removed

File: README.md [hunk 14]
```diff
@@ -1,50 +1,1323 @@
...
+| Description | String (optional)     | The description for the zone.                                |
```
Comment (sukhwinder20) [inline README.md]: Add **NOTE**: Ensure that there are no spaces

**Review by sukhwinder20 — APPROVED** [review]

**Review by ashraf-vazeer — APPROVED** [review]

**Pull request merged**